.kokoro/autorelease.sh

```diff
@@ -27,63 +27,26 @@ git config core.fileMode false
 git config user.name "$github_user"
 git config user.email "$github_email"
 
-# Download, generate, build and pack all generated libraries
-# Build support libraries in case the latest support library version isn't yet on nuget.
-
-./BuildSupport.sh
-
-./BuildGenerated.sh --onlydownload
-
-# Delete broken discovery docs here.
-# E.g. rm -f DiscoveryJson/streetviewpublish_v1.json
-
-# For more details go/dotnet-apiary-blocked-generation
+# Fetch all discovery docs into tmp/DiscoveryJson, copying new/updated ones
+# into DiscoveryJson. The tmp/ApisToGenerate.txt file lists the Discovery
+# docs that are updated.
+./UpdateDiscovery.sh tmp/ApisToGenerate.txt
+
+# If we actually want to generate everything, just list all the Discovery docs.
+if [ "$FORCE_ALL" == "true" ]
+then
+  # Note that this form of ls ends up with the file containing
+  # DiscoveryJson/xyz.json rather than just xyz.json.
+  ls DiscoveryJson/*.json > tmp/ApisToGenerate.txt
+fi
 
-# b/299933892 because of hierarchical resources
-# "nodes/{node}/nodes/{node}"
-rm -f DiscoveryJson/sasportal_v1alpha1.json
-rm -f DiscoveryJson/prod_tt_sasportal_v1alpha1.json
-# Known exception for library generation
-rm -f DiscoveryJson/apigee_v1.json
-# Name clashes because of some request's service parameter.
-rm -f DiscoveryJson/metastore_v1alpha.json
-rm -f DiscoveryJson/metastore_v1beta.json
-rm -f DiscoveryJson/metastore_v1.json
-# Investigating with Auth team
-rm -f DiscoveryJson/identitytoolkit_v1.json
-rm -f DiscoveryJson/identitytoolkit_v2.json
-rm -f DiscoveryJson/identitytoolkit_v3.json
-# Name clashes because of some response ETag field.
-rm -f DiscoveryJson/contentwarehouse_v1.json
-# b/299569133 method.request.type instead of method.request.$ref
-# type is not a supported field in method.request.
-rm -f DiscoveryJson/integrations_v1alpha.json
-rm -f DiscoveryJson/integrations_v1.json
-# b/299567447 method.request.type instead of method.request.$ref
-# type is not a supported field in method.request.
-rm -f DiscoveryJson/datalineage_v1.json
-# b/299985033 Because on non-AIP compliant RPC.
-rm -f DiscoveryJson/policysimulator_v1alpha.json
-rm -f DiscoveryJson/policysimulator_v1beta.json
+# Generate all APIs listed in tmp/NewOrUpdated.txt
+./GenerateApis.sh @tmp/ApisToGenerate.txt
```

Comment:
It's a shame the ./GenerateApis.sh cannot write to a file the paths to the projects it just generated.

Reply:
Could you maybe get those from looking at the folders that have been touched? Because then we can get rid of the "translate discovery.json to project path" utility.

Reply:
I'm trying to get away from anything that implicit. It does make it easier to build the generated code for a particular API as well, mind you. I'll keep having a think about it. (And I'll fix the comment in the above, too...)